lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -79,34 +80,42 @@ class SROA_Helper {
 public:
   // Split V into AllocaInsts with Builder and save the new AllocaInsts into Elts.
   // Then do SROA on V.
-  static bool DoScalarReplacement(Value *V, std::vector<Value *> &Elts,
-                                  Type *&BrokenUpTy, uint64_t &NumInstances,
-                                  IRBuilder<> &Builder, bool bFlatVector,
-                                  bool hasPrecise, DxilTypeSystem &typeSys,
-                                  const DataLayout &DL,
-                                  SmallVector<Value *, 32> &DeadInsts,
-                                  DominatorTree *DT);
-
-  static bool DoScalarReplacement(GlobalVariable *GV, std::vector<Value *> &Elts,
-                                  IRBuilder<> &Builder, bool bFlatVector,
-                                  bool hasPrecise, DxilTypeSystem &typeSys,
-                                  const DataLayout &DL,
-                                  SmallVector<Value *, 32> &DeadInsts,
-                                  DominatorTree *DT);
+  static bool DoScalarReplacement(
+      Value *V, std::vector<Value *> &Elts, Type *&BrokenUpTy,
+      uint64_t &NumInstances, IRBuilder<> &Builder, bool bFlatVector,
+      bool hasPrecise, DxilTypeSystem &typeSys, const DataLayout &DL,
+      SmallVector<Value *, 32> &DeadInsts, DominatorTree *DT,
+      DenseMap<const Function *, DenseMap<const BasicBlock *, unsigned>>
```

Comment:
We could really use a typedef for `DenseMap<const Function *, DenseMap<const BasicBlock *, unsigned>>`.  Perhaps `RPOTFunctionMap`?  And `RPOTIndexMap` for `DenseMap<const BasicBlock *, unsigned>`?  Open to better naming suggestions of course.

Reply:
Of course, this could be wrapped in a simple class with a getter that retrieves or creates the RPOTIndexMap for a function.  This would nicely encapsulate the block to index map construction.

Reply:
Updated.